pkg/cloud/services/eks/roles.go

```diff
@@ -204,6 +204,14 @@ func (s *NodegroupService) reconcileNodegroupIAMRole() error {
 	}
 
 	policies := NodegroupRolePolicies()
+	if len(s.scope.ManagedMachinePool.Spec.RoleAdditionalPolicies) > 0 {
```

Comment:
Can we have a `nil` exception check at `s.scope.ManagedMachinePool`

Reply:
This shouldn't be nil but looking at the creation of the scope we don't explicitly check for nil on the managed machine pool. I'd favor the nil check at the time the scope is created instead
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/scope/managednodegroup.go#L62

wdyt @shivi28 @abhinavnagaraj  ?

Reply:
Nil check is probably not required because we return if managedMachinePool is missing -- https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/3949f5376410f694defdf0ffb257a27835d7f9e5/exp/controllers/awsmanagedmachinepool_controller.go#L93
Let me add it during scope creation.

Reply:
Great, thanks.